### PR TITLE
feat(mac): Add support for "ISO" keyboard layouts

### LINKED ISSUE #2601: feat(mac): Support 102nd key
**Is your feature request related to a problem? Please describe.**
The "102nd" key is missing from the Keyman for Mac On Screen Keyboard. This key needs to be dynamically added when it is present (either with the KVK show102ndKey flag or when the system reports that the key is present -- to research if this is possible).

Reported at https://community.software.sil.org/t/german-ibm-keyboard-on-mac/2948

See oskview.m.

**Product context**
 - OS: macOS 10.x

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #2601.

This PR adds support for the 102nd key as found on European layouts. It does not however reshape the Enter key and move the BKSLASH key, as that is a much bigger job; functionally this makes it possible to press the 102nd key, which was the biggest gap.

image

and without the extra key:

image

### CODE REVIEW

File: mac/Keyman4Mac/Keyman4Mac/AppDelegate.m [hunk 1]
```diff
@@ -376,214 +376,5 @@ - (NSArray *)KVKFiles {
     
     return kvkFiles;
 }
-/*
```
Comment (mcdurdin) [inline mac/Keyman4Mac/Keyman4Mac/AppDelegate.m:379]: Just deleting related junk code on the way

File: mac/KeymanEngine4Mac/KeymanEngine4Mac/KME/MacVKCodes.h [hunk 2]
```diff
@@ -74,6 +74,7 @@
 #define MVK_ENTER           0x24
 
 /* Key Row 4 */
+#define MVK_OEM102          0x34
```
Comment (mcdurdin) [inline mac/KeymanEngine4Mac/KeymanEngine4Mac/KME/MacVKCodes.h:77]: These codes appear to be arbitrary

File: mac/KeymanEngine4Mac/KeymanEngine4Mac/KME/KMBinaryFileFormat.h [hunk 3]
```diff
@@ -182,6 +182,11 @@ struct COMP_KEY {
 
 /* KVK file format definitions */
 
+#define KVKH_102                0x01
+#define KVKH_DISPLAY_UNDERLYING 0x02
```
Comment (jahorton) [inline mac/KeymanEngine4Mac/KeymanEngine4Mac/KME/KMBinaryFileFormat.h:186]: I see that these aren't the only definitions using these values; `#define KVKH_BITMAP` is also assigned `0x01` just a few lines later.

I assume that they belong to mutually-exclusive sets, but those sets aren't exactly explicit within the file.  Maybe a comment on the range / permissible values of each set?

File: mac/KeymanEngine4Mac/KeymanEngine4Mac/KME/OnScreenKeyboard/OSKView.m [hunk 4]
```diff
@@ -159,7 +186,22 @@ - (NSArray *)oskLayout {
                          [[OSKKey alloc] initWithKeyCode:MVK_QUOTE caption:@"'" scale:1.0],
                          [[OSKKey alloc] initWithKeyCode:MVK_ENTER caption:@"↵" scale:1.75], nil];
         
-        NSArray *row4 = [NSArray arrayWithObjects:
+        NSArray *row4 = [self use102ndKey] ?
+            [NSArray arrayWithObjects:
```
Comment (jahorton) [inline mac/KeymanEngine4Mac/KeymanEngine4Mac/KME/OnScreenKeyboard/OSKView.m:190]: Is there really no way to reuse most of the defined array? This feels pretty WET.

**Comment by darcywong00** [comment]
Is the underlying key for the 102nd supposed to be <kbd>/</kbd> or <kbd><</kbd>?

I see <kbd>/</kbd> appears next to both <kbd>shift</kbd> keys.

File: mac/KeymanEngine4Mac/KeymanEngine4Mac/KME/KMBinaryFileFormat.h (hunk 3 above)
Comment (mcdurdin) [reply mac/KeymanEngine4Mac/KeymanEngine4Mac/KME/KMBinaryFileFormat.h:186]: It's `KVKK_BITMAP` 😉  but will add comments

File: mac/KeymanEngine4Mac/KeymanEngine4Mac/KME/OnScreenKeyboard/OSKView.m (hunk 4 above)
Comment (mcdurdin) [reply mac/KeymanEngine4Mac/KeymanEngine4Mac/KME/OnScreenKeyboard/OSKView.m:190]: `NSArray` isn't mutable but I could probably construct it with two separate arrays and concatenate. Agree this is WET but it's also easy to maintain ;-)

File: mac/KeymanEngine4Mac/KeymanEngine4Mac/KME/OnScreenKeyboard/OSKView.m [hunk 5]
```diff
@@ -159,7 +186,22 @@ - (NSArray *)oskLayout {
...
+                         [[OSKKey alloc] initWithKeyCode:MVK_LEFT_SHIFT caption:@"⇧" scale:1.25],
+                         [[OSKKey alloc] initWithKeyCode:MVK_OEM102 caption:@"/" scale:1.0],
```
Comment (mcdurdin) [inline mac/KeymanEngine4Mac/KeymanEngine4Mac/KME/OnScreenKeyboard/OSKView.m:192]: Should be `\`. oops.

**Review by darcywong00 — APPROVED** [review]
lgtm

**Review by jahorton — APPROVED** [review]

**Pull request merged**